position_velocity_effort_gain_controller/CMakeLists.txt

```diff
@@ -0,0 +1,117 @@
+cmake_minimum_required(VERSION 3.5)
+project(position_velocity_effort_gain_controller)
+
+# Default to C++14
+if(NOT CMAKE_CXX_STANDARD)
+  set(CMAKE_CXX_STANDARD 14)
+endif()
+
+if(CMAKE_COMPILER_IS_GNUCXX OR CMAKE_CXX_COMPILER_ID MATCHES "Clang")
+  add_compile_options(-Wall -Wextra -Wpedantic)
+endif()
+
+#1. find dependencies
+  find_package(ament_cmake REQUIRED)
+  find_package(controller_interface REQUIRED)
+  find_package(hardware_interface REQUIRED)
+  find_package(ros2_hardware_interface_bolt REQUIRED)
+  find_package(pluginlib REQUIRED)
+  find_package(rclcpp REQUIRED)
+  find_package(rclcpp_lifecycle REQUIRED)
+  find_package(realtime_tools REQUIRED)
+  find_package(std_msgs REQUIRED)
+
+
+#2. compile directive for a shared library providing the cpp file as the source
+add_library(${PROJECT_NAME}
+  SHARED
+  src/position_velocity_effort_gain_controller.cpp
+)
+
+#3.
```

Comment:
Adding comments is a very good idea.

The line below is adding the repository's `include` directory in the set of directories where headers are looking for. This happens in the code with `#include` cpp macro.